daffodil-propgen/src/main/scala/org/apache/daffodil/propGen/TunableGenerator.scala

```diff
@@ -196,16 +189,23 @@ class PrimitiveTunable(name: String, schemaType: String, schemaDefault: String)
     case "xs:boolean" => "Boolean"
     case "xs:int" =>  "Int"
     case "xs:long" => "Long"
+    case "xs:string" if (name == "tempFilePath") => "java.io.File"
     case "xs:string" => "String"
   }
 
   private val scalaDefault = schemaType match {
-    case "xs:string" if (name != "tempFilePath") => "\"" + schemaDefault + "\""
+    case "xs:string" if (name == "tempFilePath") => "new java.io.File(System.getProperty(\"java.io.tmpdir\"))"
+    case "xs:string" => "\"" + schemaDefault + "\""
     case _ => schemaDefault
   }
 
   override val scalaDefinition = s"""val ${name}: ${scalaType} = ${scalaDefault}"""
-  override val scalaConversion = s"""case "${name}" => this.copy(${name} = value.to${scalaType})"""
+  override val scalaConversion = {
+		if (name == "tempFilePath")
+			s"""case "${name}" => this.copy(${name} = new java.io.File(value))"""
+		else
+			s"""case "${name}" => this.copy(${name} = value.to${scalaType})"""
+	}
```

Comment:
This now has a special case for both scalaDefault and scalaConversion, which are the only two things that need to be implemented in TunableBase. Maybe instead just create a new TunableBase unique for tempFilePath that has the appropriate scalaDefault/Conversion and instantiate it in writeGeneratedCode above? Isolates all this special casing to a single class and leves PrimitiveTunable generic.